engine/runtime/src/test/java/org/enso/compiler/ExecCompilerTest.java

```diff
@@ -231,6 +248,45 @@ public void inlineReturnSignatureWithoutArguments() throws Exception {
     assertEquals("Function-return syntax can be used with 0 arguments", 23, result.asInt());
   }
 
+  /**
+   * This test demonstrates a slightly un-intuitive, but apparently needed by our rules, behaviour of `->` with ascriptions:
+   * 1. for `foo a:Integer -> Integer` it is interpreted as foo (a:Integer) -> Integer - i.e. a function taking an Integer and returning an Integer.
+   * 2. for `foo a : Integer -> Integer`, this results in a compile error currently.
+   */
+  @Test
+  public void weirdReturnTypeSignature1() throws Exception {
```

Comment:
Why is the test called `weird`? I'd expect that something fails then, but apparently it works. Calling `nonIntuitiveForRadek` ;-)